### PR TITLE
Add Complex dtype and sparse cholesky decomposition into sparse module.

### PR DESCRIPTION
**Pull request by chengts95**
This PR adds Complex64, Complex128 dtype and sparse cholesky decomposition into sparse module.

### CODE REVIEW

**Comment by chengts95** [comment]
It should fix issue #932 #743.

File: cupy/sparse/compressed.py [hunk 1]
```diff
@@ -99,8 +99,8 @@ def __init__(self, arg1, shape=None, dtype=None, copy=False):
         else:
             dtype = numpy.dtype(dtype)
 
-        if dtype != 'f' and dtype != 'd':
-            raise ValueError('Only float32 and float64 are supported')
+            # if dtype != 'f' and dtype != 'd':
```
Comment (unnonouno) [inline cupy/sparse/compressed.py]: Check complex dtypes. It is required for int dtype.

File: cupy/sparse/coo.py [hunk 2]
```diff
@@ -111,8 +111,8 @@ def __init__(self, arg1, shape=None, dtype=None, copy=False):
         else:
             dtype = numpy.dtype(dtype)
 
-        if dtype != 'f' and dtype != 'd':
-            raise ValueError('Only float32 and float64 are supported')
+        # if dtype != 'f' and dtype != 'd':
```
Comment (unnonouno) [inline cupy/sparse/coo.py]: ditto

File: cupy/sparse/linalg/solve.py [hunk 3]
```diff
@@ -71,3 +71,64 @@ def lsqr(A, b):
...
+    The function solves ``Ax = b``. Given two-dimensional matrix ``A`` is
+    decomposed into ``Q * R``.
+
+    Args:
+        A (cupy.ndarray or cupy.sparse.csr_matrix): The input matrix with
+            dimension ``(N, N)``
+        b (cupy.ndarray): Right-hand side vector.
+
+    Returns:
+        ret (tuple): Its length must be ten. It has same type elements
```
Comment (unnonouno) [inline cupy/sparse/linalg/solve.py]: ``tuple: ...``

**Comment by chengts95** [comment]
Ok I will split the PR first, then think about the type checking, in fact I find that some people are using type inferring in Cython to do data type check, may be it would be faster than checking the dtype in pure python.

**Comment by unnonouno** [comment]
All your PRs are merged. I'll close this original one. Thank you very much!

**Comment by huanvo88** [comment]
Hi so does this PR address cholesky decomposition for sparse matrices? I need it for cupy

**Pull request closed without merging**